snuba/api.py

```diff
@@ -216,24 +212,22 @@ def parse_and_run_query(validated_body, timer):
 
     from_clause = u'FROM {}'.format(table)
 
-    # For now, we only need FINAL if:
-    #    1. The project has been marked as needing FINAL (in redis) because of recent
-    #       replacements (and it affects too many groups for us just to exclude
-    #       those groups from the query)
-    #    OR
-    #    2. the force_final setting = 1
-    needs_final, exclude_group_ids = get_projects_query_flags(project_ids)
-    if len(exclude_group_ids) > max_group_ids_exclude:
-        # Cap the number of groups to exclude by query and flip to using FINAL if necessary
-        needs_final = True
-        exclude_group_ids = []
-
-    used_final = False
-    if force_final == 1 or (force_final is None and needs_final):
+    turbo = body.get('turbo', False)
+    if not turbo:
```

Comment:
Would it make sense to make this logic dataset specific and imported only by datasets that support merge and deletion ?

Reply:
Yep, that's my longer-term plan here. 👍Just wanted to get this out of the way first so it's a bit more encapsulated and that change will be a move rather than changing the behavior and moving the code at the same time.